lib/workflow/factory.js

```diff
@@ -5,15 +5,15 @@ var Node = require('../node/node');
 var async = require('async');
 var debug = require('../util/debug')('factory');
 
-var typeNodeMap = nodes.reduce(function(typeNodeMap, node) {
+var DEFAULT_TYPE_NODE_MAP = nodes.reduce(function(typeNodeMap, node) {
     typeNodeMap[node.TYPE] = node;
     return typeNodeMap;
 }, {});
 
-function Factory(user, databaseService) {
+function Factory(user, databaseService, typeNodeMap) {
     this.user = user;
     this.databaseService = databaseService;
-    this.typeNodeMap = typeNodeMap;
+    this.typeNodeMap = typeNodeMap || DEFAULT_TYPE_NODE_MAP;
```

Comment:
Don't know if it's better to assign or merge the new `typeNodeMap` with the default one.

On one hand, we are able to reuse the Factory in the tests and we're able to make tests for test nodes and existing nodes, on the other hand, this could be a bit messy.

What do you think?